app/src/main/java/com/github/amusinamaria/repository/data/JsonLoad.kt

```diff
@@ -0,0 +1,75 @@
+package com.github.amusinamaria.repository.data
+
+import android.content.Context
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.withContext
+import kotlinx.serialization.decodeFromString
+import kotlinx.serialization.json.Json
+
+private val jsonFormat = Json { ignoreUnknownKeys = true }
+
+private suspend fun loadGenres(context: Context): List<Genre> = withContext(Dispatchers.IO) {
+    val data = readAssetFileToString(context, "genres.json")
+    parseGenres(data)
+}
+
+internal fun parseGenres(data: String): List<Genre> {
+    val jsonGenres = jsonFormat.decodeFromString<List<Genre>>(data)
+    return jsonGenres.map { Genre(id = it.id, name = it.name) }
```

Comment:
You get a List of Genres from decodeFromString(). You don't need to convert

Reply:
Done